### PR TITLE
Add io.github.flattool.Warehouse

### PR DESCRIPTION
**Pull request by heliguy4599**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

`flatpak-spawn` needed to manage flatpaks on the host.
Flatpak directories needed for app icons and managing user data.

### CODE REVIEW

File: io.github.flattool.Warehouse.json [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                "*"
+            ]
+        },
+        {
+            "name" : "warehouse",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "dir",
```
Comment (hfiguiere) [inline io.github.flattool.Warehouse.json:49]: this won't work.

Comment (heliguy4599) [reply io.github.flattool.Warehouse.json]: Sorry, I uploaded the wrong manifest. I have edited it now.

**Comment by heliguy4599** [comment]
Linter PR: https://github.com/flathub/flatpak-builder-lint/pull/175

**Comment by heliguy4599** [comment]
For clarity, Warehouse needs spawn permissions to manage flatpaks on the host (the main purpose of the app is to manage installed flatpaks and their user data)

**Comment by AtiusAmy** [comment]
Woah this is so cool and need to be merged !!!!!!

**Pull request closed without merging**